src/Interpreters/inplaceBlockConversions.cpp

```diff
@@ -360,7 +378,7 @@ void fillMissingColumns(
         {
             /// We must turn a constant column into a full column because the interpreter could infer
             /// that it is constant everywhere but in some blocks (from other parts) it can be a full column.
-            res_columns[i] = type->createColumnConstWithDefaultValue(num_rows)->convertToFullColumnIfConst();
+            res_columns[i] = createColumnWithDefaultValue(*requested_column->getTypeInStorage(), requested_column->getSubcolumnName(), num_rows);
```

Comment:
I haven't look at this PR thoroughly because the unit tests are broken.
Looks like errors are related to the column type changes.